static/js/controllers/user-language-modal.js

```diff
@@ -0,0 +1,64 @@
+  var inboxServices = angular.module('inboxServices');
+
+  inboxServices.factory('UserLanguageModal', ['$uibModal',
+    function($uibModal) {
+      return function(enabledLocales) {
+        var modalInstance = $uibModal.open({
+          templateUrl: 'templates/modals/user_language.html',
+          controller: 'UserLanguageModalCtrl',
+          resolve: {
+            enabledLocales: function () {
+              return enabledLocales;
+            }
+          }
+        });
+        return modalInstance.result;
+      };
+    }
+  ]);
+
+angular.module('inboxControllers').controller('UserLanguageModalCtrl',
+  ['enabledLocales', '$log', '$q', '$scope', 'Session', 'SetLanguage', '$translate', '$uibModalInstance', 'UpdateUser',
+  function(enabledLocales, $log, $q, $scope, Session, SetLanguage, $translate, $uibModalInstance, UpdateUser) {
```

Comment:
For this instance, I agree we actually don't need it. Removed.
For later, injecting params might come up again, it's kinda how the uibmodal lib is used.